server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -292,7 +292,10 @@ public void applyClusterState(ClusterChangedEvent event) {
             RepositoriesMetaData newMetaData = state.getMetaData().custom(RepositoriesMetaData.TYPE);
 
             // Check if repositories got changed
-            if ((oldMetaData == null && newMetaData == null) || (oldMetaData != null && oldMetaData.equals(newMetaData))) {
+            if ((oldMetaData == null && newMetaData == null) || (oldMetaData != null && oldMetaData.equalsIgnoreGenerations(newMetaData))) {
```

Comment:
Should we check whether the generations actually changed to avoid going through all the InProgress items?

Reply:
> Should we check whether the generations actually changed to avoid going through all the InProgress items?

I wonder if that's worth it. Then we have to do all this checking ahead of time even on repos that don't look at the cluster state. Seems more efficient doing it the current way (we can optimize away all the looking at `SnapshotsInProgress` etc. in a follow-up as well, so the apply method becomes really short in the end in `BlobStoreRepository`)?

> Are you following this up with a change that looks at the actual repositories instead of event.previousState()?

Yea ... def. let's do that. That's been a long time TODO (I think we talked about it first in 2018 ...) sorry for not dealing with it earlier.